src/server/scripts/Pet/pet_dk.cpp

```diff
@@ -137,6 +145,81 @@ class npc_pet_dk_guardian : public CreatureScript
         }
 };
 
+enum DancingRuneWeapon
+{
+    DATA_INITIAL_TARGET_GUID    = 1,
+
+    EVENT_ENGAGE_VICTIM         = 1
+};
+
+struct npc_pet_dk_rune_weapon : public ScriptedAI
+{
+    npc_pet_dk_rune_weapon(Creature* creature) : ScriptedAI(creature) { }
+
+    void InitializeAI() override
+    {
+        // Prevent early victim engage
+        me->SetReactState(REACT_PASSIVE);
+        _events.ScheduleEvent(EVENT_ENGAGE_VICTIM, 1s);
+    }
+
+    void IsSummonedBy(Unit* summoner) override
+    {
+        DoCast(summoner, SPELL_COPY_WEAPON, true);
+        DoCast(summoner, SPELL_DK_RUNE_WEAPON_MARK, true);
+        DoCast(me, SPELL_DK_DANCING_RUNE_WEAPON_VISUAL, true);
+        DoCast(me, SPELL_FAKE_AGGRO_RADIUS_8_YARD, true);
+
+        me->GetThreatManager().RegisterRedirectThreat(SPELL_DK_DANCING_RUNE_WEAPON, summoner->GetGUID(), 100);
+    }
+
+    void SetGUID(ObjectGuid const& guid, int32 id) override
+    {
+        if (id == DATA_INITIAL_TARGET_GUID)
+        {
+            _targetGUID = guid;
+            if (Unit *target = ObjectAccessor::GetUnit(*me, _targetGUID))
```

Comment:
```cpp
src/server/scripts/Pet/pet_dk.cpp:181:32:
 fatal error: use of undeclared identifier 'ObjectAccessor'
            if (Unit *target = ObjectAccessor::GetUnit(*me, _targetGUID))
                               ^
```
This is a nopch error, so the solution is most likely to add
```cpp
#include "ObjectAccessor.h"
```
in the list of includes on top of the file.